### ISSUE TITLE
Adding a way to set the value of `cling::utils::out`

### ISSUE THREAD

**Issue by MostafaNanticock** [root-project/cling#530]
We have a project that uses Cling to add the ability to do some actions using c++ at runtime.
currently the only way for us to show the user the result of executing a line such as `int a = 5` is through `std::cout`.
The problem is, we are already using `std::cout` to print out many things like when logging is enabled in the application.</br>
Take this output for example:
```
[Info] started interactive c++ console
[cling]$ int a = 5
(int) 0
[cling]$ auto pvw = createNewPlotViewWindow()
[Info] Creeating a new PlotViewWindow...
[Info] Initializing OpenGL...
[Info] Using OpenGL version 4.6
[Info] OpenGL vendor: NVIDIA Corporation
[Info] OpenGL renderer: Quadro T2000/PCIe/SSE2
[Info] is OpenGLES: false
[Info] OpenGL version: 4.6
[Info] OpenGL options: 
    depthBufferSize 24,
    redBufferSize 8,
    greenBufferSize 8,
    blueBufferSize 8,
    alphaBufferSize 8,
    stencilBufferSize 8,
    samples 0,
    swapBehavior DoubleBuffer,
    swapInterval 1,
    colorSpace sRGBColorSpace
[Info] OpenGL initialization finished
[Info] Window initialization done
(PlotViewWindow *) 0x7dff598e91
[cling]$
```
and as you can see it becomes so frustrating especially when the user tries to scroll back through the terminal to check his previous inputs.
And after thorough examination of the Cling source code, it appears that the best solution would be for us to implement our own `llvm::raw_os_ostream` that we can then use as the default `cling::utils::outs`, `cling::utils::errs` and `cling::utils::logs`,
This would also allow us to integrate the logging library that we are using with Cling directly, so that it can simply log any cling message or error at real-time.

I don't know if there is any reason behind the choice for not having this functionality, and it's fairly easy to implement.
The simple solution that i can think of right now would be to add three functions in the form of:
```cpp
namespace cling::utils {
    void setOuts(llvm::raw_ostream &newOuts);
    void setErrs(llvm::raw_ostream &newErrs);
    void setLog(llvm::raw_ostream &newLog);
}
``` 
I'd be happy to submit a pull request with this simple addition if any of the maintainers agreed on this proposed change.

**Comment by ferdymercury** [comment]
Did you try with any of the redirection commands?

```
   .> <filename>		- Redirect command to a given file
      '>' or '1>'		- Redirects the stdout stream only
      '2>'			- Redirects the stderr stream only
      '&>' (or '2>&1')		- Redirects both stdout and stderr
      '>>'			- Appends to the given file
```

> I'd be happy to submit a pull request with this simple addition if any of the maintainers agreed on this proposed change.

adding @hahnjo or @vgvassilev to the loop.

**Comment by MostafaNanticock** [comment]
Yes, I know about the ability to redirect output to files, But the way it's implemented is by redirecting the c++ `std::cout` and `std::cerr` streams to a given file. Which doesn't solve the original problem.

Consider running the following example after redirecting stdout to a given file:
```cpp
[cling]$ std::cout << "Hello" << std::endl;
```
This will cause the word 'Hello' to be outputted to the file, which isn't what we want.

What we want is to have some way to only redirect the outputs of cling itself. Whether it's a warning, an error or just the immediate output of executing any statement without adding the semicolon to the end.
For example, when you execute this line:
```
[cling]$ std::cout << "Hello" << std::endl
```
the output would be:
```
Hello
(basic_ostream<char, std::char_traits<char> > &) @0x7ff98e524fc0
```
where the first line is just the result of appending stuff to `std::cout` and the second output is the result of executing the line inside cling

We want to be able to only redirect Cling output

**Comment by ferdymercury** [comment]
Thanks for the clarification. If you consider proposing a PR, it should be done directly here: https://github.com/root-project/root/pulls under interpreter/cling subdirectory.

**Comment by MostafaNanticock** [comment]
Should i raise an issue there first in order to make sure that this modification doesn't conflict with anything, or it's ok to go there and male the pull request directly?

**Comment by ferdymercury** [comment]
The issue can stay here, no need to transfer.
You can already put the Pull Request directly there, and you can write "Fixes https://github.com/root-project/cling/issues/530" as second line of the commit.
